@types/automerge/index.d.ts

```diff
@@ -203,53 +158,61 @@ declare module 'automerge' {
     // no public methods or properties
   }
 
+  // A change request, sent from the frontend to the backend
+  interface Request {
+    requestType: RequestType
+    actor: string
+    seq: number
+    time: number
+    message: string
+    version: number
+    ops: Op[]
+  }
+
   interface Change {
     message?: string
-    requestType?: RequestType
     actor: string
     seq: number
-    deps: Clock
+    deps: Hash[]
     ops: Op[]
-    diffs?: Diff[]
   }
 
   interface Op {
     action: OpAction
     obj: UUID
-    key?: string
-    value?: any
+    key: string | number
+    insert?: boolean
```

Comment:
For me it is really hard to reason about a optional bool. 
A bool should either true or false. If you really need to express three states, I would prefer a enum with named states.

Reply:
This is not the only place in the type specification where optional bool is used. Maybe we could clean this up if possible

Reply:
Isn't this a side-effect of writing something in idiomatic JavaScript? Certainly good advice for TypeScript code, but encoding these states as `0, 1, 2` would be pretty opaque to the JavaScript consumer, and using strings would bloat the in-memory cost. In practice, I imagine you'd use a falsy check here, so `undefined` would be equivalent to `false`. I assume this is still just two states (`falsy`, `true`) rather than three (`undefined`, `false`, `true`) in any meaningful sense.

Reply:
It's not tristate, it's just a boolean that defaults to false when absent. I don't know how to declare that in TypeScript. In 02271f082803e57ccc2c2d7890fc5ec49ab08f3f I've changed operations such that they always have an `insert` property that is explicitly true or false.